docs/DeveloperGuide.adoc

```diff
@@ -153,6 +163,101 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+During execution, the List command creates the intended `Context` and updates the `Model` with it, along with the appropriate `FilteredList`. As `MainWindow` tracks both `FilteredList`, JavaFX will automatically re-render the contained card entries when the respective list undergoes structural changes.
+
+The updated `ContextType` of the `Model` is then returned to the `MainWindow` UI controller via the `CommandResult`, where it is used to switch the content displayed.
+
+Given below is an example usage scenario of a user intending to view the list of activities.
+
+Step 1. The user launches the `SplitWiser` application for the first time, which defaults to displaying the list of contacts.
+
+Step 2. The user executes the command string `list a/` to prompt the app to display the full list of activities.
+
+The following sequence diagram demonstrates the role of the respective components.
+
+image::ViewSequenceDiagram.png[]
+
+==== Design Considerations
+
+===== Aspect: representing the current context state
+
+* **Alternative 1 (current choice):** Creating an immutable `Context` class as a state container to store the appropriate information.
+** Pros: Lightweight; context state can be easily passed between components that require access, as it is encapsulated entirely within `Context`.
+** Cons: Harder to implement.
+
+* **Alternative 2:** Storing the `Context` state in a designated central component, e.g. the `ModelManager`, and exposing relevant methods to other components.
+** Pros: Trivial to implement.
+** Cons: Tightly couples all components that require access to the context state to the central component, complicating testing.
+
+// tag::invitedisinvite[]
+=== [Proposed] Invite/Disinvite feature
+==== Proposed Implementation
+
+The invite/disinvite mechanism is facilitated by `Activity`.
+It extends `AddressBook` with an `ActivityBook`, stored internally as an `activityList`. Additionally, it implements the following operations:
+
+* `Activity#invite()` -- Invites a person to the activity.
+* `Activity#disinvite()` -- Disinvites a person from the activity.
+
+These operations are exposed in the `Activity` class as `Activity#invite()` and `Activity#disinvite()` respectively.
+
+Given below is an example usage scenario and how the invite/disinvite mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The `SplitWiser` will be initialized with the initial address book and activity book state.
+
+Step 2. The user executes `activity t/breakfast p/David` command to create an activity named breakfast in the activity book.
+
+Step 3. The user executes `invite p/Louis ...` to invite more people into the current activity. The `invite` command calls `Activity#invite()`, inviting more participants into the activity. The activityList will then be modified which is stored in the ActivityBook.
```

Comment:
Formatting for activityList inconsistent - the one here doesn't use a code quote but elsewhere it is placed in a code quote as `activityList`.